src/pages/dashboard/panels/UpcomingAnime.tsx

```diff
@@ -1,20 +1,27 @@
-import React, { useState } from 'react';
+import React, { useMemo, useState } from 'react';
 import { useSelector } from 'react-redux';
 
 import ShokoPanel from '@/components/Panels/ShokoPanel';
 import TransitionDiv from '@/components/TransitionDiv';
 import { useGetDashboardAniDBCalendarQuery } from '@/core/rtkQuery/splitV3Api/dashboardApi';
+import { useGetSettingsQuery } from '@/core/rtkQuery/splitV3Api/settingsApi';
 import DashboardTitleToggle from '@/pages/dashboard/components/DashboardTitleToggle';
 import EpisodeDetails from '@/pages/dashboard/components/EpisodeDetails';
+import { initialSettings } from '@/pages/settings/SettingsPage';
 
 import type { RootState } from '@/core/store';
 
 const UpcomingAnime = () => {
   const layoutEditMode = useSelector((state: RootState) => state.mainpage.layoutEditMode);
 
   const [showAll, setShowAll] = useState(false);
-  const localItems = useGetDashboardAniDBCalendarQuery({ showAll: false });
-  const items = useGetDashboardAniDBCalendarQuery({ showAll: true });
+
+  const settingsQuery = useGetSettingsQuery();
+  const settings = useMemo(() => settingsQuery.data ?? initialSettings, [settingsQuery.data]);
```

Comment:
Dependency array should be `[settingsQuery]`.
If you use `settingsQuery.data`, useMemo will be able to do nothing and it will change value on every render

Reply:
Should ideally be `settingsQuery.requestId` but linter will complain